starfyre/parser.py

```diff
@@ -18,17 +18,41 @@ def extract_functions(obj):
 
 
 class ComponentParser(HTMLParser):
-    # this is the grammar for the parser
-    # we need cover all the grammar rules
+    """
```

Comment:
@SuelenKarbivnychyy , I have added some explanation. To summarise what we have done so far.